### PR TITLE
use the same way that is used in mattermost/docs to generate the preview site

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
use the same way that is used in mattermost/docs to generate the preview site

for fork, PRs need to add the `preview environment label to kick the build

The infrastructure needed to this is here and already created: https://git.internal.mattermost.com/ops/infrastructure/aws/aws-core/-/merge_requests/2

### CODE REVIEW

**Review by cwarnermm — APPROVED** [review]
Great addition!

File: .github/workflows/preview-env-fork.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+          cat pr.txt
+          echo "::set-output name=id::$(<pr.txt)"
+      - name: Check label
+        if: ${{ github.event.workflow_run.conclusion == 'success' }}
+        uses: actions/github-script@v4
+        id: label-checker
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            const query = `query($owner:String!, $name:String!, $pr:Int!) {
```
Comment (metanerd) [inline .github/workflows/preview-env-fork.yml:31]: Maybe we should call a javascript file here called is-label.js? WDYT?

File: .github/workflows/preview-env.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        with:
+          aws_key_id: ${{ secrets.AWS_KEY_ID }}
+          aws_secret_access_key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws_bucket: ${{ secrets.AWS_BUCKET }}
+          source_dir: ./dist/html
+          destination_dir: ${{ github.event.number }}
+
+      - name: Add comment to PR
+        if: github.event.pull_request.head.repo.full_name == github.repository
+        uses: peter-evans/create-or-update-comment@v1
```
Comment (metanerd) [inline .github/workflows/preview-env.yml:41]: Not sure I like this from a sec perspective.

File: .github/workflows/preview-env.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        with:
+          hugo-version: '0.74.2'
+
+      - name: Build Previews
+        if: github.event.pull_request.head.repo.full_name == github.repository
+        run: |
+          rm -rf ./dist
+          hugo -F -s site --baseURL http://mattermost-dev-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/${{ github.event.number }}/ --destination ../dist/html
+
+      - uses: shallwefootball/s3-upload-action@master
```
Comment (metanerd) [inline .github/workflows/preview-env.yml:29]: Not sure I like this from a sec perspective.

File: .github/workflows/preview-env-fork.yml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        if: ${{ github.event.workflow_run.conclusion == 'success' && steps.label-checker.outputs.result == 'true' }}
+        with:
+          aws_key_id: ${{ secrets.AWS_KEY_ID }}
+          aws_secret_access_key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws_bucket: ${{ secrets.AWS_BUCKET }}
+          source_dir: ./dist/html
+          destination_dir: ${{ steps.pr.outputs.id }}
+
+      - name: Add comment to PR
+        uses: peter-evans/create-or-update-comment@v1
```
Comment (metanerd) [inline .github/workflows/preview-env-fork.yml:90]: Not sure I like this from a sec perspective.

File: .github/workflows/preview-env-fork.yml [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+          hugo-version: '0.74.2'
+          # extended: true
+
+      - name: Build Previews
+        if: ${{ github.event.workflow_run.conclusion == 'success' && steps.label-checker.outputs.result == 'true' }}
+        run: |
+          rm -rf ./dist
+          hugo -F -s site --baseURL http://mattermost-dev-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/ --destination ../dist/html
+
+      - uses: shallwefootball/s3-upload-action@master
```
Comment (metanerd) [inline .github/workflows/preview-env-fork.yml:79]: Not sure I like this from a sec perspective.

**Review by metanerd — COMMENTED** [review]
Thank you, LGTM and I have some concerns.

File: .github/workflows/preview-env.yml (hunk 2 above)
Comment (cpanato) [reply .github/workflows/preview-env.yml:41]: we can review the tagged code and evaluate it. I already did a look and did not see anything that is questinable

File: .github/workflows/preview-env.yml (hunk 3 above)
Comment (cpanato) [reply .github/workflows/preview-env.yml:29]: we can review the tagged code and evaluate it. I already did a look and did not see anything that is questinable

File: .github/workflows/preview-env-fork.yml (hunk 5 above)
Comment (cpanato) [reply .github/workflows/preview-env-fork.yml:79]: we can review the tagged code and evaluate it. I already did a look and did not see anything that is questionable

File: .github/workflows/preview-env-fork.yml (hunk 4 above)
Comment (cpanato) [reply .github/workflows/preview-env-fork.yml:90]: we can review the tagged code and evaluate it. I already did a look and did not see anything that is questionable

File: .github/workflows/preview-env-fork.yml (hunk 1 above)
Comment (cpanato) [reply .github/workflows/preview-env-fork.yml:31]: you mean instead of the variable name query replace with is-label?

Comment (metanerd) [reply .github/workflows/preview-env-fork.yml:31]: Sorry, was maybe not clear enough. I thought maybe put it in an external script like a bash script. Maybe not.

Comment (cpanato) [reply .github/workflows/preview-env-fork.yml:31]: dont need to do this, but i'm 0/5 maybe can do in a follow up if is really needed

File: .github/workflows/preview-env-fork.yml (hunk 4 above)
Comment (metanerd) [reply .github/workflows/preview-env-fork.yml:90]: I think it still could be externally retagged without our notice AFAIU. So I find this questionable. Especially if we provide these sensitive parameters.

File: .github/workflows/preview-env-fork.yml (hunk 1 above)
Comment (metanerd) [reply .github/workflows/preview-env-fork.yml:31]: Agreed. Not really blocking.

**Review by metanerd — APPROVED** [review]

**Pull request merged**